### ISSUE TITLE
`nu -i -c` does not source `config.nu`

### ISSUE THREAD

**Issue by JustinLovinger** [nushell/nushell#7442]
### Describe the bug

`nu -i -c` does not source `config.nu`.

### How to reproduce

1. Add `alias test = echo "test"` to `config.nu`.
2. Run `nu -i -c test`.

### Expected behavior

I expected `nu` to start an interactive shell with configuration from `config.nu`, run commands from `-c`, and exit after those commands finished.

### Configuration

| key                | value                                |
| ------------------ | ------------------------------------ |
| version            | 0.72.0                               |
| branch             |                                      |
| commit_hash        |                                      |
| build_os           | linux-x86_64                         |
| build_target       | x86_64-unknown-linux-gnu             |
| rust_version       | rustc 1.64.0                         |
| cargo_version      | cargo 1.64.0                         |
| pkg_version        | 0.72.0                               |
| build_time         | 1980-01-01 00:00:00 +00:00           |
| build_rust_channel | release                              |
| features           | database, default, trash, which, zip |
| installed_plugins  |                                      |

P.S. ignore `build_time`. This is NixOS.

### Additional context

You can work around the bug by adding `--config`.

I believe `env.nu` is not sourced either, but I did not test this as thoroughly.

**Comment by fdncred** [comment]
This is by design which is why we give you `--config` and `--env-config` parameters if you want this information sourced.

**Comment by fdncred** [comment]
No, because the intent of `-e` is to run a command and then remain in the repl. This was specifically engineered to start nushell, in normal fashion, but in a particular directory.

**Comment by JustinLovinger** [comment]
I see the intent, but I think it would be more intuitive if `-i` started an interactive shell with `config.nu` sourced by default, and the user could optionally source another config or no config.

**Comment by fdncred** [comment]
`nu -i` does load the config files, at least on the latest main

**Comment by JustinLovinger** [comment]
At least on 0.72.0, `nu -i` does *not* load the config if given alongside `-c`. It seems the `-i` option is silently ignored.

**Comment by fdncred** [comment]
Agreed. It seemed like combining those two flags ignores loading the config files. I'm not sure if that's by design or not.

**Comment by rgwood** [comment]
I suspect this is unintentional and the combination of `-i` and `-c` flags has not been tested before (but I did not write that code, that's just my guess).

@JustinLovinger can you provide a little context as to why you're trying to do this? That will help us prioritize the issue.

**Comment by JustinLovinger** [comment]
> @JustinLovinger can you provide a little context as to why you're trying to do this? That will help us prioritize the issue.

I use this with a script to open terminal applications in a window, as if they were non-terminal applications, `exec alacritty -t "''${PWD/#$HOME/'~'}:$*" -e $SHELL -i -c "$*" > /dev/null 2>&1 & disown`, used like `t vi path/to/file`. The script is supposed to work with alias, functions, and everything else you would expect from an interactive shell, so you can simply prepend a command with `t`. This script works with others shells I tested.

**Comment by schmitzli** [comment]
with  0.27 i was running my definitions with a command like 
`nu -c my-def-cmd`

as it is no longer loading the definitions, with the hint about -e i found this workaround
`nu -e "my-def-cmd ; exit"`

**Comment by schmitzli** [comment]
for me it is useful when triggering my commands from outside the shell, like a scheduler job or from a gui front end

**Comment by schmitzli** [comment]
oh my workaround is only working from a cmd shell...

Error: 
  Ã— Nushell launched as a REPL, but STDIN is not a TTY; either launch in a
  â”‚ valid terminal or provide arguments to invoke a script!

**Comment by schmitzli** [comment]
nu -c xxx with the optional script parameter does not seem to work also.
So for my case i'll pass my_def.nu to the nu as --config which seems to work

`nu -c my-def-cmd --config complete\path\to\my_defs.nu `

**Comment by melMass** [comment]
I think most of the behaviours explained here changed since, posting @fdncred 's explanation from discord:

## nu -c "ls"
- ✅ Always read personal plugin.nu file if it exists
- ✅ Always reads the default_env.nu file
- ⛔ Does not read any config.nu file
- ⛔ Does not read any login.nu file

## nu -l -c "ls"
- ✅ Always read personal plugin.nu file if it exists
- ✅ Always reads personal env.nu file if it exists
- ✅ Always reads personal config.nu if it exists
- ✅ Always reads personal login.nu if it exists

## nu -l -c "ls" --config foo_config.nu
- ✅ Always reads personal plugin.nu file if it exists
- ✅ Always reads personal env.nu file if it exists
- ✅ Always reads specified config file foo_config.nu
- ✅ Always reads personal login.nu if it exists

## nu -l -c "ls" --config foo_config.nu --env-config foo_env.nu
- ✅ Always read personal plugin.nu file if it exists
- ✅ Always reads specified file foo_env.nu
- ✅ Always reads specified file foo_config.nu
- ✅ Always reads personal login.nu if it exists

## nu -n -l -c "ls"
- ⛔ Does not load any plugin.nu file
- ⛔ Does not load any env.nu file
- ⛔ Does not load any config.nu file
- ⛔ Does not load any login.nu file

*2024-01-18 nushell version 0.89.1 commit 55bf4d8*

**Comment by jasha-hrp** [comment]
When I run `nu -i -c 'echo 123'` I'm not getting an interactive shell.
```
 nu -i -c 'echo 123'
123
# nushell immediately exists without dropping to an interactive shell
```

**Comment by melMass** [comment]
> ```
>  nu -i -c 'echo 123'
>  123
>  # nushell immediately exists without dropping to an interactive shell
> ```

That's expected:
```nushell
  -c, --commands <string> - run the given commands and then exit
```

You are looking for `--execute(-e)`:

```nushell
  -e, --execute <string> - run the given commands and then enter an interactive shell
```

**Comment by kgadek** [comment]
> for me it is useful when triggering my commands from outside the shell, like a scheduler job or from a gui front end

I have the same use-case and hit same issue. I got it working by using the following invocation:

```nushell
nu -l -c 'foo'
```

This does load config files and sets `NU_LIB_DIRS` correctly, runs my command defined in `env.nu`, and finally exits.

----

My confusion around this came from the documentation:

- https://www.nushell.sh/book/special_variables.html#nu:

  > `is-interactive`
  > […] Note: When started as an interactive shell, startup config files are processed. When started as a non-interactive shell, no config files are read unless explicitly called via flag.

- no flag is explicitly mentioned, but `nu --help`:

  ```nushell
  […]
  -e, --execute <string>: run the given commands and then enter an interactive shell
  -i, --interactive: start as an interactive shell
  -l, --login: start as a login shell
  ```

  No other flag seems fitting.

  From this I concluded that `-i` flag would load config files, because `nu -i -c '$nu.is-interactive'` returns `true`. That's not the case though.

----

If this is intended behaviour then allow me to suggest updating documentation to (a) clarify when config files are loaded, (b) perhaps provide information what's the idiomatic way to load config files (is it via `-l` flag or by some kind of `source-env …` magic (it didn't work for me when I tried it).
